internal/cli/deploy.go

```diff
@@ -0,0 +1,215 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"text/template"
+
+	"github.com/google/uuid"
+	libvirt "github.com/libvirt/libvirt-go"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+)
+
+const libvirtDomainDefinitionTemplate = `
```

Comment:
We have same file in `test/e2e/data/vm.xml` can we reuse it somehow?

Reply:
This is a template. to reuse `data/vm.xml` we need to move everything in some kind of builder somehow and reuse it on cli and e2e.
We can do this in a separate PR.